packages/flutter/lib/src/services/asset_resolver.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:async';
+
+import 'asset_bundle.dart';
+
+/// Base class for resolver components that transform an input into part of
+/// a resolved asset's key.
+abstract class Qualifier {
+  String resolve();
+}
+
+/// Qualifier that knows how to find the nearest match among a list of
+/// candidates, based on a list of the normative values for each candidate.
+///
+/// Breakpoints and resolutions should be provided in ascending order. The
+/// resolution is done via linear search based on the assumption that the list
+/// of candidates is short.
+class BreakpointQualifier extends Qualifier {
+  BreakpointQualifier({ this.candidates, this.values, this.value }) {
+    assert(this.candidates.length == this.values.length);
+    assert(this.candidates.length > 0);
+  }
+  final List<String> candidates;
+  final List<double> values;
+  double value;
+  String resolve() {
+    for (int i = 0; i < candidates.length-1; i++) {
```

Comment:
good practice suggests `i += 1` or `++i` since otherwise people will learn to use `i++` and eventually use it somewhere with a non-optimising compiler where it uses some additional cycles for no good reason.